app/app.go

```diff
@@ -871,6 +872,34 @@ func (app *App) ProcessProposalHandler(ctx sdk.Context, req *abci.RequestProcess
 	}, nil
 }
 
+func (app *App) BuildDependencyDag(ctx sdk.Context, txs [][]byte) (*Dag, error) {
+	// contains the latest msg index for a specific Access Operation
+	dependencyDag := Dag{
+		NodeMap:  make(map[int]DagNode),
+		EdgesMap: make(map[int][]DagEdge),
+	}
+	for txIndex, txBytes := range txs {
+		tx, err := app.txDecoder(txBytes)
```

Comment:
with this there will be two repetitive decoding for each tx (once here, once in `runMsg`). We don't need to optimize it in this PR but let's add a TODO here in case we forget